ProjectLighthouse.Servers.Website/Pages/PasswordResetRequestForm.cshtml.cs

```diff
@@ -0,0 +1,67 @@
+using JetBrains.Annotations;
+using LBPUnion.ProjectLighthouse.Configuration;
+using LBPUnion.ProjectLighthouse.Helpers;
+using LBPUnion.ProjectLighthouse.PlayerData;
+using LBPUnion.ProjectLighthouse.PlayerData.Profiles;
+using LBPUnion.ProjectLighthouse.Servers.Website.Pages.Layouts;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace LBPUnion.ProjectLighthouse.Servers.Website.Pages;
+
+public class PasswordResetRequestForm : BaseLayout
+{
+
+    public string? Error { get; private set; }
+
+    public string? Status { get; private set; }
+
+    public PasswordResetRequestForm(Database database) : base(database)
+    { }
+
+    [UsedImplicitly]
+    public async Task<IActionResult> OnPost(string username)
+    {
+
+        if (!ServerConfiguration.Instance.Mail.MailEnabled)
+        {
+            this.Error = "Email disabled in lighthouse configuration, unable to continue.";
+            return this.Page();
+        }
+
+        if (string.IsNullOrWhiteSpace(username))
+        {
+            this.Error = "The username field is required.";
+            return this.Page();
+        }
+
+        User? user = await this.Database.Users.FirstOrDefaultAsync(u => u.Username == username);
+
+        if (user == null)
+        {
+            this.Error = "User does not exist";
+            return this.Page();
+        }
+
+        PasswordResetToken token = new()
+        {
+            Created = DateTime.Now,
+            UserId = user.UserId,
+            ResetToken = CryptoHelper.GenerateAuthToken(),
+        };
+
+        string messageBody = $"Hello {user.Username}\n\n"+
+        $"Your {ServerConfiguration.Instance.Customization.ServerName} password can be reset at the following link\n" +
+        $"{ServerConfiguration.Instance.ExternalUrl}/passwordReset?token={token.ResetToken}";
+
+        SMTPHelper.SendEmail(user.EmailAddress,
+            "Password reset request", messageBody);
+
+        this.Database.PasswordResetTokens.Add(token);
+        await this.Database.SaveChangesAsync();
+
+        this.Status = $"Password reset email sent to {user.EmailAddress}.";
```

Comment:
Try not to leak user data, this could've been pretty bad lol
```suggestion
        this.Status = $"Password reset email sent.";
```

Reply:
Might be worth doing a mix of the two eg. e******@e******.o** just in case the user has multiple E-Mail addresses

Reply:
Fair enough, I also see other websites/webapps do that.

Reply:
If the community has a problem with this we can always revert it to not showing at all.

Reply:
Ok Done